index.js

```diff
@@ -1,39 +1,33 @@
-import { showItemsControlContainer } from './modules/domElements.js';
-import {
-  openModalButton,
-  showModal,
-  hideModal,
-  closeModalButton,
-} from './modules/modal.js';
-import { toggleItemsControls } from './modules/itemsControl.js';
+import { hideAddFormModal } from './modules/modal.js';
+import { hideOptionsModal } from './modules/optionsModal.js';
 import { populateItems } from './modules/populateItems.js';
 import db from './modules/db.js';
+import { exportDb } from './modules/exportDb.js';
+
+// get download button and add action
+const downloadButton = document.getElementById('backupData');
+downloadButton.addEventListener('click', exportDb);
 
 // call populate to load shopping list items
 window.onload = populateItems;
 
 /* MODAL ACTIONS */
-// When the user clicks the button, open the modal
-openModalButton.addEventListener('click', showModal);
-
-// When the user clicks on <span> (x), close the modal
-closeModalButton.addEventListener('click', hideModal);
 
 // When the user clicks anywhere outside of the modal, close it
 window.addEventListener('click', (event) => {
+  // add options modal
   if (event.target === modal) {
-    hideModal();
+    hideAddFormModal();
   }
 });
 
 // When the user presses the Escape key, close the modal
 window.addEventListener('keydown', (event) => {
   if (event.key === 'Escape') {
-    hideModal();
+    hideAddFormModal();
+    hideOptionsModal();
```

Comment:
This needs better set up. We call both of these functions and both of them return focus to their respective "open" buttons. 

That means whenever we press `esc` and the Add Item modal is open, the focus is put on the settings button.

Reply:
Fixed this by toggling an `open` class on the modal for the Add Item form. 

This way, if this is open, we'll always close that first and stop - returning focus to the add new item button, and if the settings are also open they will remain open. The add form has a z index to appear above the settings panel when open, so I think this behavior tracks. 

So - 

if both modals are open -> add item modal is on top
if esc is pressed once -> add item form modal closes (focus on the button)
if esc is pressed once more -> settings panel closes (focus on the settings button)